From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mja00 <teamnirvana2@gmail.com>
Date: Tue, 22 Jun 2021 16:42:10 -0400
Subject: [PATCH] Making sure the API version is correct


diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index 774556a62eb240da42e84db4502e2ed43495be17..2305c48772ec7ed85b73887ecb0af043d39411fa 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/io.papermc.paper/paper-api/pom.properties");
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/com.papyrus/papyrus-api/pom.properties");
         Properties properties = new Properties();
 
         if (stream != null) {
